### PR TITLE
Element/Attribute names must be case sensitive (see #433)

### LINKED ISSUE #433: Element/Attribute names must be case sensitive
XML Element/Attribute names must be case sensitive. For instance if we have this xml

`<a></A>`

For example if we set the cursor in `<a>`, document highlight should not highlight `</A>`

### PR DESCRIPTION
**Pull request by angelozerr**
This PR fix #433 and avoid using String#toLowerCase() which creates a
lot of instances of String.

### CODE REVIEW

**Comment by fbricon** [comment]
@angelozerr any noticeable performance impact?

**Comment by angelozerr** [comment]
> @angelozerr any noticeable performance impact?

not really -( I just notice that we gain some ms (very few) but editor continue to be slow. But this PR follow the XML spec for case sensitive tag

**Comment by angelozerr** [comment]
@NikolasKomonen @xorye  is there any chance that you review this simple PR please.

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLHighlightingTest.java [hunk 1]
```diff
@@ -74,8 +74,8 @@ public void selfclosed() throws BadLocationException {
 
 	@Test
 	public void caseInsensivity() throws BadLocationException {
```
Comment (xorye) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLHighlightingTest.java]: I think a better name for this test case would be `caseSensitivity` or `caseNoInsensivity`

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLRenameTest.java [hunk 2]
```diff
@@ -79,8 +79,8 @@ public void selfclosed() throws BadLocationException {
 
 	@Test
 	public void caseInsensivity() throws BadLocationException {
```
Comment (xorye) [inline org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLRenameTest.java]: I think a better name for this test case would be caseSensitivity  or `caseNoInsensivity`

**Comment by fbricon** [comment]
Tests should be renamed *caseSensitive

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLHighlightingTest.java (hunk 1 above)
Comment (angelozerr) [reply org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLHighlightingTest.java]: Good catch! I rename my test with `caseSensitive` according @fbricon suggestion

File: org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLRenameTest.java (hunk 2 above)
Comment (angelozerr) [reply org.eclipse.lsp4xml/src/test/java/org/eclipse/lsp4xml/services/XMLRenameTest.java]: Good catch! I rename my test with `caseSensitive` according @fbricon suggestion

**Comment by angelozerr** [comment]
Thanks guys for your review. I merge the PR even if travis failed with https://travis-ci.org/angelozerr/lsp4xml/builds/545792737#L930 (I don't understand why it failed?)

**Pull request merged**